kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java

```diff
@@ -126,33 +116,56 @@ private void processKieService(ClassLoader classLoader, String key, String value
                     childServices.computeIfAbsent( serviceName, k -> new ArrayList<>() )
                             .add( newInstance( classLoader, value.substring( 1 ) ) );
                 } else {
-                    services.put( serviceName, newInstance( classLoader, value ) );
+                    int priority = 0;
+                    int separatorPos = value.indexOf( ';' );
+                    if (separatorPos > 0) {
+                        priority = Integer.parseInt( value.substring( separatorPos+1 ).trim() );
+                        value = value.substring( 0, separatorPos ).trim();
+                    }
+                    services.put( priority, serviceName, newInstance( classLoader, value ) );
+                    log.debug( "Added Service " + value + " with priority " + priority );
                 }
             } catch (RuntimeException e) {
                 if (optional) {
                     log.info("Cannot load service: " + serviceName);
                 } else {
-                    System.out.println("Loading failed because " + e.getMessage());
+                    log.error("Loading failed because " + e.getMessage());
                     throw e;
                 }
             }
-            log.debug( "Adding Service {}\n", value );
```

Comment:
Can you please restore this log or add a log message inside the [`if` block](https://github.com/kiegroup/droolsjbpm-knowledge/blob/03ba485e72b2ba1796765f4e2f9e68e11bf1b2de/kie-api/src/main/java/org/kie/api/internal/utils/ServiceDiscoveryImpl.java#L115-L117) too?